frame/staking/src/lib.rs

```diff
@@ -1215,6 +1311,41 @@ pub mod pallet {
 	#[pallet::storage]
 	pub(crate) type StorageVersion<T: Config> = StorageValue<_, Releases, ValueQuery>;
 
+	// The next storage items collectively comprise the voter bags: a composite data structure
+	// designed to allow efficient iteration of the top N voters by stake, mostly. See
+	// `mod voter_bags` for details.
+	//
+	// In each of these items, voter bags are indexed by their upper weight threshold.
+
+	/// How many voters are registered.
+	#[pallet::storage]
+	pub(crate) type CounterForVoters<T> = StorageValue<_, u32, ValueQuery>;
+
+	/// Which bag currently contains a particular voter.
+	///
+	/// This may not be the appropriate bag for the voter's weight if they have been rewarded or
+	/// slashed.
+	#[pallet::storage]
+	pub(crate) type VoterBagFor<T: Config> =
+		StorageMap<_, Twox64Concat, AccountIdOf<T>, VoteWeight>;
```

Comment:
Thinking aloud: What if instead of having this map we just stored the current bag in the node itself? This would make it so reading in a node is just 1 read, instead of the 2 needed to get the node and then its bag from here

(I am guessing though I am forgetting some other access pattern where this is map is important)

Reply:
Yeah it seems like it is not used anywhere exclusively.

Reply:
@kianenigma closing bc if we move voter_bags to its own pallet this storage item would actually be useful